### PR TITLE
Porting the code to pytorch 0.4.0

### PR DESCRIPTION
**Pull request by dhwajraj**
some api syntax level changes for running the code with PYTORCH 0.4.0

### CODE REVIEW

File: src/torchcrf/__init__.py [hunk 1]
```diff
@@ -55,19 +54,19 @@ def reset_parameters(self) -> None:
...
     def __repr__(self) -> str:
         return f'{self.__class__.__name__}(num_tags={self.num_tags})'
 
     def forward(self,
-                emissions: Variable,
-                tags: Variable,
-                mask: Optional[Variable] = None,
+                emissions: torch.Tensor,
+                tags: torch.Tensor,
+                mask: Optional[torch.Tensor] = None,
```
Comment (kmkurn) [inline src/torchcrf/__init__.py]: `mask` should be `Optional[torch.ByteTensor]`.

File: src/torchcrf/__init__.py [hunk 2]
```diff
@@ -111,16 +110,16 @@ def forward(self,
                 raise ValueError('mask of the first timestep must all be on')
 
         if mask is None:
-            mask = Variable(self._new(tags.size()).fill_(1)).byte()
+            mask = torch.Tensor(self._new(tags.size()).fill_(1)).byte()
```
Comment (kmkurn) [inline src/torchcrf/__init__.py]: I think you can get rid of `torch.Tensor` here. Could you please check?

File: src/torchcrf/__init__.py [hunk 3]
```diff
@@ -111,16 +110,16 @@ def forward(self,
...
         numerator = self._compute_joint_llh(emissions, tags, mask)
         denominator = self._compute_log_partition_function(emissions, mask)
         llh = numerator - denominator
         return llh if not reduce else torch.sum(llh)
 
     def decode(self,
-               emissions: Union[Variable, torch.FloatTensor],
-               mask: Optional[Union[Variable, torch.ByteTensor]] = None) -> List[List[int]]:
+               emissions: Union[torch.Tensor, torch.FloatTensor],
+               mask: Optional[Union[torch.Tensor, torch.ByteTensor]] = None) -> List[List[int]]:
```
Comment (kmkurn) [inline src/torchcrf/__init__.py]: Since now variable and tensor are the same, emissions should be just `torch.FloatTensor` and mask should be just `torch.ByteTensor`. No need for `Union`.

File: src/torchcrf/__init__.py [hunk 4]
```diff
@@ -148,19 +147,19 @@ def decode(self,
                 f'got {tuple(emissions.size()[:2])} and {tuple(mask.size())}'
             )
 
-        if isinstance(emissions, Variable):
+        if isinstance(emissions, torch.Tensor):
             emissions = emissions.data
```
Comment (kmkurn) [inline src/torchcrf/__init__.py]: I think line 151-152 can be safely removed.

File: src/torchcrf/__init__.py [hunk 5]
```diff
@@ -148,19 +147,19 @@ def decode(self,
...
         if mask is None:
             mask = self._new(emissions.size()[:2]).fill_(1).byte()
-        elif isinstance(mask, Variable):
+        elif isinstance(mask, torch.Tensor):
             mask = mask.data
```
Comment (kmkurn) [inline src/torchcrf/__init__.py]: I think line 155-156 can also be safely removed.

File: src/torchcrf/__init__.py [hunk 6]
```diff
@@ -148,19 +147,19 @@ def decode(self,
...
 
         return self._viterbi_decode(emissions, mask)
 
     def _compute_joint_llh(self,
-                           emissions: Variable,
-                           tags: Variable,
-                           mask: Variable) -> Variable:
+                           emissions: torch.Tensor,
+                           tags: torch.Tensor,
+                           mask: torch.Tensor) -> torch.Tensor:
```
Comment (kmkurn) [inline src/torchcrf/__init__.py]: `mask` should be `torch.ByteTensor`.

File: src/torchcrf/__init__.py [hunk 7]
```diff
@@ -197,8 +196,8 @@ def _compute_joint_llh(self,
         return llh
 
     def _compute_log_partition_function(self,
-                                        emissions: Variable,
-                                        mask: Variable) -> Variable:
+                                        emissions: torch.Tensor,
+                                        mask: torch.Tensor) -> torch.Tensor:
```
Comment (kmkurn) [inline src/torchcrf/__init__.py]: I think `mask` should be `torch.ByteTensor` here as well.

**Comment by kmkurn** [comment]
This has been covered by #25; I'm closing this PR.
